client/app/scripts/superdesk-vocabularies/vocabularies.js

```diff
@@ -0,0 +1,152 @@
+/**
+ * This file is part of Superdesk.
+ *
+ * Copyright 2013, 2014 Sourcefabric z.u. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code, or
+ * at https://www.sourcefabric.org/superdesk/license
+ */
+
+(function() {
+  'use strict';
+
+  VocabularyService
+      .$inject = [ 'api', 'urls', 'session', '$upload', '$q', 'metadata' ];
+  function VocabularyService(api, urls, session, $upload, $q, metadata) {
+    var service = this;
+
+    this.vocabularies = undefined;
+
+    this.getVocabularies = function(scope) {
+      if (typeof service.vocabularies === 'undefined') {
+        api.query('vocabularies')
+            .then(function(result) {
+              service.vocabularies = result;
+              scope.vocabularies = service.vocabularies;
+            });
+      } else {
+        scope.vocabularies = service.vocabularies;
+      }
+    };
+  };
+
+  VocabularyConfigController.$inject =
+      [ '$scope', 'gettext', 'session', 'modal', 'notify', 'vocabularies' ];
+  function VocabularyConfigController($scope, gettext, session, modal, notify,
+                                      vocabularies) {
+    $scope.vocabularies = null;
+    $scope.vocabulary = null;
+
+    $scope.createVocabulary = function() {
```

Comment:
Since `metadata.values` is cached we either need to update the `metadata.values` or invalidate the `metadata.values`.